raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -3154,14 +3279,15 @@ private static <T> Registration<T> unregister(Listener<T> listener) {
     }
 
     private final class ListenerContext implements CloseListener<BatchReader<T>> {
+        private static final long STARTING_NEXT_OFFSET = -1;
```

Comment:
Does it make sense to also have a constant for 0 (log start) ?

Reply:
Done. Named it SMALLEST_LOG_OFFSET since Kafka already uses "log start offset" and that has a different meaning.